backend/src/main/java/de/neuefische/backend/appUser/AppUserService.java

```diff
@@ -14,17 +15,21 @@ public class AppUserService {
     private final AppUserRepository appUserRepository;
     private final BCryptPasswordEncoder passwordEncoder;
 
-    //private int newUserIban = 1;
-    public AppUser create (AppUser appUser) {
+
+    public AppUser create(AppUser appUser) {
         Optional<AppUser> existingAppUser = findByUsername(
                 appUser.getUsername()
         );
 
         if (existingAppUser.isPresent()) {
             throw new ResponseStatusException(HttpStatus.CONFLICT);
         }
-       // String iban = "DE99 9009 9009 0000 0" + newUserIban++;
-        // appUser.setIban(iban);
+
+        Optional <AppUser> maxAccountNumber =
+                appUserRepository.findTopByOrderByAccountNumberDesc();
+        int accountNumber = maxAccountNumber.map(AppUser::getAccountNumber).orElse(0) + 1;
```

Comment:
Isn't this method referencing itself?

Reply:
as mentioned above its a function from the library and what the code does it takes out the max. account number and always adds +1 to it. So when the User is created every user has a unique account number

Reply:
Alrighty than! Sounds good
*Check*